packages/nimble-components/src/rich-text/mention-list-box/index.ts

```diff
@@ -0,0 +1,294 @@
+                    this.scrollOptionIntoView();
+                    return true;
+                }
+                return false;
+            }
+            case keyArrowUp: {
+                if (!event.shiftKey && this.hasAnySelectableOption) {
+                    this.listBox.selectPreviousOption();
+                    this.scrollOptionIntoView();
+                    return true;
+                }
+                return false;
+            }
+            case keyEscape: {
+                this.setOpen(false);
+                return false;
+            }
+            default: {
+                return false;
+            }
+        }
+    }
+
+    /**
+     * @public
+     */
+    public clickHandler(e: MouseEvent): boolean {
+        // eslint-disable-next-line @typescript-eslint/no-unnecessary-type-assertion
+        const captured = (e.target as HTMLElement).closest(
+            'option,[role=option]'
+        ) as ListOption | null;
+
+        if (!captured || captured.disabled) {
+            return false;
+        }
+        this.activateMention({
+            href: captured.value,
+            displayName: captured.innerText
+        } as MentionDetail);
+        return true;
+    }
+
+    public updateMentionExtensionConfig(mentionExtensionConfig?: MentionExtensionConfiguration[]): void {
+        this.mentionExtensionConfig = mentionExtensionConfig;
+        this.setActiveMappingConfigs();
+        this.filterOptions();
+    }
+
+    public onMention(props: SuggestionProps): void {
+        this.suggestionProps = props;
+        this.setActiveCharacter(props.text.slice(0, 1));
+        this.filter = props.query;
+        this.setAnchorElement(props.decorationNode as HTMLElement);
+        this.setOpen(true);
+        void this.selectFirstOptionIfValidOptionExists();
+    }
+
+    public close(): void {
+        this.setOpen(false);
+    }
+
+    /**
+     * @internal
+     */
+    public activeCharacterChanged(): void {
+        this.setActiveMappingConfigs();
+    }
+
+    /**
+     * @internal
+     */
+    public filterChanged(): void {
+        this.filterOptions();
+    }
+
+    /**
+     * @internal
+     */
+    public childItemsChanged(): void {
+        void this.selectFirstOptionIfValidOptionExists();
+    }
+
+    /**
+     * @internal
+     */
+    public anchorElementChanged(prev: HTMLElement, next: HTMLElement): void {2
+        if (prev) {
+            this.intersectionObserver.unobserve(prev);
+        }
+        if (this.region && this.anchorElement) {
+            this.region.anchorElement = this.anchorElement;
+            this.region.update();
+            this.intersectionObserver.observe(next);
+        }
+    }
+
+    private setActiveCharacter(activeCharacter: string): void {
+        this.activeCharacter = activeCharacter;
+    }
+
+    private setAnchorElement(anchorElement: HTMLElement): void {
+        this.anchorElement = anchorElement;
+    }
```

Comment:
Seems like we should just inline these two functions, since they're each only called once.

Reply:
Updated the logic